source_code/src/AES/aes.c

```diff
@@ -22,10 +22,13 @@
 #define F(x)   (((x)<<1) ^ ((((x)>>7) & 1) * 0x1b))
 #define FD(x)  (((x) >> 1) ^ (((x) & 1) ? 0x8d : 0))
 
-#define BACK_TO_TABLES
 #ifdef BACK_TO_TABLES
 
-const uint8_t sbox[256] __attribute__ ((__progmem__)) = {		// forward s-box
+#if defined(__AVR__)
+const uint8_t sbox[256] __attribute__ ((__progmem__)) = {        // forward s-box
+#else
+const uint8_t sbox[256] = {        // forward s-box

```

Comment:
I might be mistaken here... but shouldn't you declare an empty array here as if you don't the c routine will populate the array values with the the ones fetched from flash?